packages/plugin-phone/src/web-rtc-media.js

```diff
@@ -265,7 +268,20 @@ const WebRTCMedia = AmpState.extend({
     });
 
     this.on(`change:localMediaStream`, () => {
-      // TODO introspect the stream and update audio/video accordingly
+      if (!this.peer) {
+        return;
+      }
+      const streams = this.peer.getLocalStreams();
+      if (!streams.includes(this.localMediaStream)) {
+        streams.forEach((stream) => {
+          removeStream(this.peer, stream);
+        });
+        addStream(this.peer, this.localMediaStream);
+        this.set({
+          sendingAudio: getLocalMediaStatus(`audio`, this.peer),
+          sendingVideo: getLocalMediaStatus(`video`, this.peer)
+        });
+      }
```

Comment:
Is this what happens when someone mutes and then unmutes video?
Is there a chance that this would fire after a developer's event listener for the `event:localMediaStream` event and thus not be able to grab the new updated stream?